Applications/Utils/FileConverter/NetCdfConverter.cpp

```diff
@@ -0,0 +1,696 @@
+        if (dim == 3)
+            std::cout << "(p)rism or (h)exahedron: ";
+        std::string type("");
+        std::getline(std::cin, type);
+        checkExit(type);
+        if (dim == 2)
+        {
+            if (type != "t" && type != "q" && type != "tri" && type != "quad" &&
+                type != "triangle" && type != "quatliteral")
+                continue;
+            if (type == "t" || type == "tri" || type == "triangle")
+                return MeshLib::MeshElemType::TRIANGLE;
+            else
+                return MeshLib::MeshElemType::QUAD;
+        }
+
+        if (dim == 3)
+        {
+            if (type != "p" && type != "h" && type != "prism" &&
+                type != "hex" && type != "hexahedron")
+                continue;
+            if (type == "p" || type == "prism")
+                return MeshLib::MeshElemType::PRISM;
+            else
+                return MeshLib::MeshElemType::HEXAHEDRON;
+        }
+    }
+    return t;
+}
+
+bool multFilesSelectionLoop(
+    std::pair<std::size_t, std::size_t> const& time_bounds)
+{
+    std::size_t const n_time_steps(time_bounds.second - time_bounds.first + 1);
+    std::cout << "\nThe selection includes " << n_time_steps
+              << " time steps.\n";
+    std::cout << "0. Save data in " << n_time_steps
+              << " mesh files with one scalar array each.\n";
+    std::cout << "1. Save data in one mesh file with " << n_time_steps
+              << " scalar arrays.\n";
+    std::size_t ret = parseInput("Select preferred method: ", 2, false);
+    return (ret == 0) ? true : false;
+}
+
+std::string getIterationString(std::size_t i, std::size_t max)
+{
+    std::size_t const max_length(std::to_string(max).length());
+    std::string const current_str(std::to_string(i));
+    return std::string(max_length - current_str.length(), '0') + current_str;
+}
+
+double getResolution(NcFile const& dataset, NcVar const& var)
+{
+    std::size_t const dim_idx = var.getDimCount() - 1;
+    auto const bounds = getBoundaries(getDimVar(dataset, var, dim_idx));
+    return fabs(bounds.second - bounds.first) /
+           static_cast<double>(var.getDim(dim_idx).getSize());
+}
+
+GeoLib::RasterHeader createRasterHeader(
+    NcFile const& dataset, NcVar const& var,
+    std::array<std::size_t, 4> const& dim_idx_map,
+    std::vector<std::size_t> const& length, bool is_time_dep)
+{
+    MathLib::Point3d const origin = getOrigin(dataset, var, dim_idx_map, is_time_dep);
+    double const res = getResolution(dataset, var);
+    std::size_t n_dims = var.getDimCount();
+    std::size_t temp_offset = (is_time_dep) ? 1 : 0;
+    std::size_t z_length =
+        (n_dims - temp_offset == 3) ? length[dim_idx_map.back()] : 1;
+    return {length[dim_idx_map[0 + temp_offset]],
+            length[dim_idx_map[1 + temp_offset]],
+            z_length, origin, res, -9999};
+}
+
+std::size_t getLength(NcVar const& var, bool const is_time_dep,
```

Comment:
It's returning the length of the total data array as well as the length of each individual dimension. In VTK, using parameters to return data happens all the time. What do you propose?

Reply:
To return both values, especially if the vector is of fixed size, it could be a `std::array` saving memory allocation on heap.
```suggestion
std::tuple<std::size_t, std::array<std::size_t, 4>> getLength(NcVar const& var, bool const is_time_dep)
```

Reply:
The netcdf function where the argument is used is expecting a vector, though.

Reply:
good, then let's make it a vector:
```suggestion
std::tuple<std::size_t, std::vector<std::size_t>> getLength(NcVar const& var, bool const is_time_dep)
```

Reply:
Yes, obviously. I just pointed out the vector thing because you've mentioned allocation on the heap.